android/src/main/java/net/singular/react_native/SingularBridgeModule.java

```diff
@@ -42,32 +42,7 @@ public String getName() {
     }
 
     @ReactMethod
-    public void init(String apiKey, String secret, String customUserId, int sessionTimeout) {
-        config = new SingularConfig(apiKey, secret);
-
-        if (customUserId != null) {
-            config.withCustomUserId(customUserId);
-        }
-
-        if (sessionTimeout >= 0) {
-            config.withSessionTimeoutInSec(sessionTimeout);
-        }
-
-        Singular.init(reactContext, config);
-    }
-
-    @ReactMethod
-    public void initWithSingularLink(String apiKey, String secret, String customUserId, int sessionTimeout) {
-        config = new SingularConfig(apiKey, secret);
-
-        if (customUserId != null) {
-            config.withCustomUserId(customUserId);
-        }
-
-        if (sessionTimeout >= 0) {
-            config.withSessionTimeoutInSec(sessionTimeout);
-        }
-
+    public void init(SingularConfig config) {
```

Comment:
I think it's a bit problematic to send the config as SingularConfig object.
Let's verify but i think RN only accepts a whitelisted types of arguments.
Maybe let's treat the config as a JSON string and parse it the same is we did for Obj-C.
We can sync and do it together if you want.